sherpa/sim/sample.py

```diff
@@ -452,6 +466,9 @@ def get_sample(self, fit, *, myscales=None, num=1):
             it is calculated from the fit object.
         num : int, optional
             The number of samples to return.
+        rng : numpy.random.Generator, numpy.random.RandomState, or None, optional
```

Comment:
```suggestion
        rng : `numpy.random.Generator`, `numpy.random.RandomState`, or None, optional
```

Reply:
OK, enough of these comment. First, please try one and see if that actually works for make the links html in the RTD and then a global "search and replace" on the entire codebase is going to be way more efficient since the line comes up many, many times, sometimes with no and sometimes with numpy. (Also wit hand without optional, see some of my other comments). That still only two search-and-replace operations. Why do I do the work that a computer can do better and faster?

Reply:
To be honest, I am not sure it's needed. I don't understand the rules of when the expansion works, but here is the output I get for the `sherpa.sim.sample.NormalParameterSampleFromScaleMatrix.get_sample` routine

![image](https://github.com/sherpa/sherpa/assets/224638/8d6efc55-46bb-45b3-8dbd-41ea11234ed9)

and hee is the docstring that created this:

```
 489   │ class NormalParameterSampleFromScaleMatrix(ParameterSampleFromScaleMatrix):
 490   │     """Use a normal distribution to sample parameters (correlated),
 491   │ 
 492   │     The parameters are drawn from a normal distribution based on the
 493   │     parameter errors, and include the correlations between the
 494   │     parameters. The errors will be generated from the fit object or
 495   │     specified directly as a covariance matrix.
 496   │ 
 497   │     """
 498   │ 
 499   │     def get_sample(self, fit, *, mycov=None, num=1, rng=None):
 500   │         """Return the parameter samples.
 501   │ 
 502   │         Parameters
 503   │         ----------
 504   │         fit : sherpa.fit.Fit instance
 505   │             This defines the thawed parameters that are used to generate
 506   │             the samples, along with any possible error analysis.
 507   │         mycov : 2D numpy array or None, optional
 508   │             The covariance matrix to use. If None then it is
 509   │             calculated from the fit object.
 510   │         num : int, optional
 511   │             The number of samples to return.
 512   │         rng : numpy.random.Generator, numpy.random.RandomState, or None, optional
 513   │             If set, the generator is used to create the random numbers. If
 514   │             not set then the legacy numpy RandomState instance is used.
 515   │ 
 516   │         Returns
 517   │         -------
 518   │         samples : 2D numpy array
 519   │             The array is num by npar size, where npar is the number of
 520   │             free parameters in the fit argument.
 521   │ 
 522   │         """
 523   │         vals = numpy.array(fit.model.thawedpars)
 524   │         cov = self.scale.get_scales(fit, mycov)
 525   │         return random.multivariate_normal(rng, vals, cov, size=int(num))
 526   │ 
```

so you can see that (at least on my computer, unfortunately the RTD version for this PR is currently broken as it had some internal error, but hopefully it'll work soon after I fix a conflict and some of the comments so we can check there).

Reply:
(e.g. `sherpa.fit.Fit` hasn't been picked up here as a link; is the "intersphnix inventory" changes you made a while back that makes it pick up the `numpy.xxx` symbols?

Reply:
So, if the text says `numpy.<symbol>` then it should work (for valid symbols) but if the text is `np.<symbol>` then it won't (it becomes something that looks like it could be a link [bold fixed font] but is not guaranteed to be a link).

[this comment is for my own benefit]

Reply:
So my current idea is not to make these changes, as we don't need them, but let me know if I'm missing a use case where they are useful (or some other reason)

Reply:
Here's the RTD version showing the same behavior: https://sherpa--1735.org.readthedocs.build/en/1735/mcmc/api/sherpa.sim.sample.NormalParameterSampleFromScaleMatrix.html#sherpa.sim.sample.NormalParameterSampleFromScaleMatrix.get_sample (just to check it wasn't something magic about my system)

Reply:
I'll close the other "make them links" comments and leave this open as a reminder

Reply:
> is the "intersphnix inventory" changes you made a while back that makes it pick up the numpy.xxx symbols

I think so.

Reply:
Maybe I was more successful than I had thought. It's a good think to get rid of the extra backticks, but I can't say I understand when they are required either. I think in the astropy package that we pull in the name "numpy" might be special-cased to work without backticks.

Reply:
It's not ideal that we have some special-cased text here, but I just don't want to add the backticks here if I can help it ;-)